modules/caddyhttp/reverseproxy/streaming.go

```diff
@@ -96,8 +96,8 @@ func (h Handler) flushInterval(req *http.Request, res *http.Response) time.Durat
 		return -1 // negative means immediately
 	}
 
-	// for h2 and h2c upstream streaming data to client (issue #3556)
-	if req.ProtoMajor == 2 && res.ContentLength == -1 {
+	// for h2 and h2c upstream streaming data to client (issue #3556,#3606)
+	if req.ProtoMajor == 2 && res.ContentLength == -1 && req.Header.Get("Accept-Encoding") == "identity" {
```

Comment:
Can we abstract these conditions into a function instead of repeating it? And also split the conditions onto separate lines by `&&`.
```suggestion
	if req.ProtoMajor == 2 &&
		res.ContentLength == -1 &&
		req.Header.Get("Accept-Encoding") == "identity" {
```

Reply:
Yes, that would look better.